types/apiResponse.ts

```diff
@@ -0,0 +1,18 @@
+export enum StatusMessageType {
+  Error = 0,
+  Warning = 1,
+  Information = 2,
+  Success = 3,
+}
+
+export interface StatusMessage {
+  type: StatusMessageType;
+  message: string;
+}
+
+export interface ApiResponse<T> {
+  payload: T | EmptyPayload;
+  messages: StatusMessage[];
+}
+
+export type EmptyPayload = Record<string, never>;
```

Comment:
do you need to specify `ApiResponse<T | EmptyPayload>` in every file, considering you already made `payload: T | EmptyPayload`?